### PR TITLE
ID-286 ID-287 Remove modernizr and stop including lodash in the bundle

### PR DESCRIPTION
**Pull request by matmannion**
* Remove Modernizr dependency, adds a new FeatureDetect module with just the bare-bones, mostly using well-supported browser JS like matchMedia and CSS.supports
* Add babel-plugin-lodash and stop exposing all of lodash as `window._`, which significantly reduces the bundle size

This is most noticeable in the size difference of `id7-bundle.js`:

id7-bundle.js  | Size | Size (gzipped)
----------|-------|---------------------
Before | 264KB | 83.6KB
After | 197KB | 60.8KB

### CODE REVIEW

File: js/feature-detect.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
+/* eslint-env browser */
+const matchMedia = window.matchMedia || window.msMatchMedia;
+
+export default class FeatureDetect {
+  constructor() {
+    this.tests = {};
+  }
+
+  static mq(query) {
+    const mql = matchMedia(query);
```
Comment (halfninja) [inline js/feature-detect.js]: Check `matchMedia` is defined for IE9 and below, just so they don't error out?

Comment (flamingelk) [reply js/feature-detect.js]: Or at the top fallback to a function that just returns false, as that's what we do after this.

Comment (matmannion) [reply js/feature-detect.js]: Good call, I definitely intended to do that but then forgot

File: js/id7-standalone.js [hunk 2]
```diff
@@ -1,8 +1,71 @@
...
+try {
+  svgFilters = 'SVGFEColorMatrixElement' in window && SVGFEColorMatrixElement.SVG_FECOLORMATRIX_TYPE_SATURATE === 2;
+} catch (e) {
+  // Nothing to do here
+}
+detect.addTest('svgfilters', svgFilters);
+
+// WebP detection
+const webpImages = {
+  webp: 'data:image/webp;base64,UklGRiQAAABXRUJQVlA4IBgAAAAwAQCdASoBAAEAAwA0JaQAA3AA/vuUAAA=',
```
[the remark this answers is not in the document]
Comment (matmannion) [reply js/id7-standalone.js]: They have, this is a good point - I think I'll add a CSP section to the documentation listing what we'd expect is a "minimal" CSP, including My Warwick etc.

**Pull request merged**